tests/integration_tests/datasource/api_tests.py

```diff
@@ -0,0 +1,137 @@
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import json
+from unittest.mock import Mock, patch
+
+import pytest
+
+from superset import db, security_manager
+from superset.connectors.sqla.models import SqlaTable
+from superset.dao.exceptions import DatasourceTypeNotSupportedError
+from tests.integration_tests.base_tests import SupersetTestCase
+
+
+class TestDatasourceApi(SupersetTestCase):
+    def get_virtual_dataset(self):
+        return (
+            db.session.query(SqlaTable)
+            .filter(SqlaTable.table_name == "virtual_dataset")
+            .one()
+        )
+
+    @pytest.mark.usefixtures("app_context", "virtual_dataset")
+    def test_get_column_values_ints(self):
+        self.login(username="admin")
+        table = self.get_virtual_dataset()
+        rv = self.client.get(f"api/v1/datasource/table/{table.id}/column/col1/values/")
+        self.assertEqual(rv.status_code, 200)
+        response = json.loads(rv.data.decode("utf-8"))
+        for val in range(10):
+            assert val in response["result"]
+
+    @pytest.mark.usefixtures("app_context", "virtual_dataset")
+    def test_get_column_values_strs(self):
+        self.login(username="admin")
+        table = self.get_virtual_dataset()
+        rv = self.client.get(f"api/v1/datasource/table/{table.id}/column/col2/values/")
+        self.assertEqual(rv.status_code, 200)
+        response = json.loads(rv.data.decode("utf-8"))
+        for val in ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]:
+            assert val in response["result"]
+
+    @pytest.mark.usefixtures("app_context", "virtual_dataset")
+    def test_get_column_values_floats(self):
+        self.login(username="admin")
+        table = self.get_virtual_dataset()
+        rv = self.client.get(f"api/v1/datasource/table/{table.id}/column/col3/values/")
+        self.assertEqual(rv.status_code, 200)
+        response = json.loads(rv.data.decode("utf-8"))
+        for val in [1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 1.9]:
+            assert val in response["result"]
+
+    @pytest.mark.usefixtures("app_context", "virtual_dataset")
+    def test_get_column_values_nulls(self):
+        self.login(username="admin")
+        table = self.get_virtual_dataset()
+        rv = self.client.get(f"api/v1/datasource/table/{table.id}/column/col4/values/")
+        self.assertEqual(rv.status_code, 200)
+        response = json.loads(rv.data.decode("utf-8"))
+        self.assertEqual(response["result"], [None])
```

Comment:
nit: all the above tests could be DRYer if we use `@pytest.mark.parametrize` (optional)

Reply:
Hadn't used that before, pretty neat.  However, I couldn't get it to play nice with it being a method on a class/using `self`.  I checked, and all the other tests in the codebase using that are standalone functions.  The duplication isn't too bad here, so opting to just leave as-is